lib/client/models/webVitals.js

```diff
@@ -0,0 +1,124 @@
+import { Tracker } from 'meteor/tracker';
+import { onCLS, onFCP, onFID, onINP, onLCP, onTTFB } from 'web-vitals';
+import { getClientArchVersion } from '../../common/utils';
+import { Ntp } from '../../ntp';
+import { unwrapDynamicImport } from '../hijack/wrapDynamicImport';
+import { unwrapMethodCall } from '../hijack/wrapMethodCall';
+import { unwrapSubscription } from '../hijack/wrapSubscription';
+import { getBrowserInfo, getClientArch } from '../utils';
+import { unWrapLogin, wrapLogin } from '../hijack/wrapLogin';
+const average = arr => arr.reduce( ( p, c ) => p + c, 0 ) / arr.length || 0;
+
+export class WebVitalsModel {
+  startTime = Ntp._now();
+  connectionTime = 0;
+  loggedIn = 0;
+  importTime = [];
+  methods = [];
+  subs = [];
+  queue = new Set();
+
+  addToQueue (metric) {
+    this.queue.add({ metric: metric.value, matricName: metric.name });
+  }
+
+  flushQueue () {
+    if (this.queue.size > 0) {
+      Kadira.send(this._buildPayload([...this.queue]), '/client-metrics');
+      this.queue.clear();
+    }
+  }
+
+  startTracking () {
+    const bindedAddToQueue = this.addToQueue.bind(this);
+    onFCP(bindedAddToQueue);
+    onFID(bindedAddToQueue);
+    onINP(bindedAddToQueue);
+    onLCP(bindedAddToQueue);
+    onTTFB(bindedAddToQueue);
+    onCLS(bindedAddToQueue);
+
+    Tracker.autorun((computation) => {
+      const {connected} = Meteor.status();
+      if (connected) {
+        this.connectionTime = Ntp._now() - this.startTime;
+        computation.stop();
+      }
+    });
+
+
+    // startup hooks run after the page is loaded
+    Meteor.startup(() => {
+      wrapLogin();
+
+      // wait until all startup hooks called (they're called synchronously all at once on the client)
+      setTimeout(() => {
+        /* calling unwrap before the document readyState is complete
+         is not the best for react based apps as it's not guaranteed the components are mounted
+         and hooks are called making the subscription */
+        document.onreadystatechange = function () {
+          if (document.readyState === 'complete') {
+            /* stop tracking new dynamic imports/methods/publications */
+            unwrapDynamicImport();
+            unwrapSubscription();
+            unwrapMethodCall();
+            unWrapLogin();
+          }
+        };
+      });
+    });
+    // Report all available metrics whenever the page is backgrounded or unloaded.
+    addEventListener('visibilitychange', () => {
+      if (document.visibilityState === 'hidden') {
+        this.flushQueue();
+      }
+    });
```

Comment:
This might fire too often, such as when the user switches to another tab while waiting for the page to load.

There doesn't seem to be any good solutions. One option might be to use two approaches:
- track the number of dynamic imports, method calls, publications, etc. and once Meteor.startup runs and there are no more it is waiting on, send the metrics
- if the pagehide event fires earlier, then send the metrics immediately and indicate the page didn't finish loading. We might need to use the `beacon` api to ensure the request finishes.